components/kubernetes/component.go

```diff
@@ -0,0 +1,99 @@
+package kubernetes
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type Importable interface {
+	Key() string
+	Import(output auto.OutputMap) error
+}
+
+// The generic component resource that be created by any provider (kind, EKS, etc.)
+// We could apply the same thing for VM with AWS VMs, Azure VMs, and so on
+type KubernetesCluster struct {
+	pulumi.ResourceState
+	exportName string
+
+	KubeConfig pulumi.StringOutput `export:"kubeConfig"`
+}
+
+// The target type that's going to be deserialized on the other side.
+// We could unit test the matching if required.
+// The E stands for exported, probably can find something better
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
Agree, can be confused with `Exporter`. Maybe `KubernetesClusterDeserialiser`